action.yml

```
@@ -25,33 +30,103 @@ runs:
+      run: |
+        if [ -n "$(ls -A '${{ env.R_LIB_PATH }}' 2>/dev/null)" ]; then
+          echo "cache-hit=true" >> $GITHUB_ENV
+        else
+          echo "cache-hit=false" >> $GITHUB_ENV
+        fi
+      shell: bash
+
     - name: Install R packages (DESCRIPTION-specified deps)
       uses: r-lib/actions/setup-r-dependencies@v2
+      with:
+        extra-packages: |
+          atime
+          ggplot2
+          directlabels
+          cumstats=?ignore
+        cache: false
+
+    - name: Download libgit2 artifact
+      id: download-libgit2
+      uses: dawidd6/action-download-artifact@v10
+      with:
+        name: libgit2-cache
+        path: /tmp
+        if_no_artifact_found: ignore
+
+    - name: Check for libgit2 cache hit
+      id: libgit2-hit
+      run: |
+        if [ -f /tmp/libgit2-cache.tar.gz ]; then
+          echo "libgit2-cache-hit=true" >> $GITHUB_ENV
+        else
+          echo "libgit2-cache-hit=false" >> $GITHUB_ENV
+        fi
+      shell: bash
+
+    - name: Restore libgit2 from artifact
+      if: env.libgit2-cache-hit == 'true'
+      run: |
+        echo "Restoring libgit2 from artifact..."
+        sudo tar -xzf /tmp/libgit2-cache.tar.gz -C /
+        sudo ldconfig
+      shell: bash
 
-    - name: Set up the required system dependency libgit2
+    - name: Install libgit2 system dependency
+      if: env.libgit2-cache-hit == 'false'
       run: |
         sudo apt-get update -y && sudo apt-get install -y cmake libssl-dev
         git clone --branch v1.6.4 --depth 1 https://github.com/libgit2/libgit2.git
         cd libgit2 && mkdir build && cd build
-        cmake .. -DCMAKE_INSTALL_PREFIX=/usr
+        cmake .. -DCMAKE_INSTALL_PREFIX=/usr/local
         make && sudo make install
         cd ../..
+        echo "Creating libgit2 cache tarball..."
+        mkdir -p /tmp
+        sudo tar -czf /tmp/libgit2-cache.tar.gz \
+          /usr/local/lib/libgit2* \
+          /usr/local/include/git2* 2>/dev/null || true
```

Redirecting to `/dev/null` and forcing a `true` exit code (with the logical or) is a shortcut to masking errors and losing visibility when something goes wrong in this step and further along the line subsequently.